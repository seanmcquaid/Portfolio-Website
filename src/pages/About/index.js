import React from 'react';
import { P, H1, Headshot } from 'components';
import trumpetHeadshot from 'images/trumpetHeadshot.jp2';
import styled, { keyframes } from 'styled-components';

const About = () => (
  <AboutPageContainer id="aboutPage" data-testid="aboutPage">
    <Header>
      <H1 id="aboutPageHeader" data-testid="aboutPageHeader">
        About Me
      </H1>
    </Header>
    <Main>
      <Headshot
        src={trumpetHeadshot}
        alt="trumpetHeadshot"
        id="bioHeadshot"
        data-testid="bioHeadshot"
      />
      <Article>
        <P id="bioParagraphOne" data-testid="bioParagraphOne">
          Originally from New York, I moved to Atlanta roughly three years ago.
          My professional background is in music and now, Software Engineering.
          I have been playing the trumpet for over 20 years and have a
          Bachelor’s and Master’s degree in Music. After I moved to Atlanta, I
          quickly realized that I wanted a day job that would allow me to
          utilize my creative problem solving skills from music.
        </P>
        <P id="bioParagraphTwo" data-testid="bioParagraphTwo">
          I accidentally discovered programming one night while scrolling on
          Facebook and seeing that a musician friend of mine from high school
          recently landed a job at Google. Upon investigating more and seeing
          that his job title was a Software Engineer, I quickly came to
          understand why he decided to pursue programming. That night was when I
          wrote my first lines of HTML/CSS and I have loved it ever since! After
          six months of self studying, I decided to enroll in DigitalCrafts, a
          coding bootcamp based in Atlanta. Shortly after graduating, I was
          lucky enough to land my first Software Engineering role at Chick-fil-A
          where I worked on various parts of the CFAOne App. I am currently
          serving as a React developer on our Personal Web Ordering project
          which is being used as our online ordering platform for both domestic
          and international restaurants!
        </P>
        <P id="bioParagraphThree" data-testid="bioParagraphThree">
          It's very important to me that I continue to grow my skills and learn
          different specializations within Software Engineering so I can be a
          well informed engineer. My ultimate career goal is to be a Software
          Architect and make enterprise wide technical decisions that will help
          everyone in the company. I truly believe that having a wider
          perspective will inevitably lead to me making better decisions for the
          organization that I have the privelege to serve.
        </P>
      </Article>
    </Main>
  </AboutPageContainer>
);

const fadeInAnimation = keyframes`
    0% { 
        opacity: 0; 
    } 
    100% { 
        opacity: 1; 
    } 
`;

const AboutPageContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  transition: 0.5s;
  animation: ${fadeInAnimation} ease 1s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  @media (max-width: 975px) {
    justify-content: flex-start;
  }
`;

const Header = styled.header`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0.5rem;
`;

const Main = styled.main`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  @media (max-width: 975px) {
    flex-direction: column;
  }
`;

const Article = styled.article`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 36rem;
`;

export default About;
