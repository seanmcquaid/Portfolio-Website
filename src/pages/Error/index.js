import React from 'react';
import styled, { keyframes } from 'styled-components';
import { HeaderText, SubHeaderText } from 'components';
import { Link } from 'react-router-dom';

const Error = () => (
  <ErrorPageContainer>
    <Header>
      <HeaderText>Error</HeaderText>
    </Header>
    <Main>
      <SubHeaderText>
        There was a problem going to the page you typed in. Please go to the{' '}
        <HomePageLink to="/" title="Home Page Link">
          Home Page
        </HomePageLink>{' '}
        and check out my portfolio!
      </SubHeaderText>
    </Main>
  </ErrorPageContainer>
);

const fadeInAnimation = keyframes`
    0% { 
        opacity: 0; 
    } 
    100% { 
        opacity: 1; 
    } 
`;

const ErrorPageContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  transition: 0.5s;
  animation: ${fadeInAnimation} ease 1s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  @media (max-width: 975px) {
    justify-content: flex-start;
  }
`;

const Header = styled.header``;

const Main = styled.main``;

const HomePageLink = styled(Link)`
  text-decoration: none;
  color: #ffffffb6;
  &:hover {
    transition: 0.5s;
    color: #00000096;
    border-bottom: 2px solid #00000096;
  }
`;

export default Error;
