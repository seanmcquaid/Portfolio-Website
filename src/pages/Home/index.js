import React from 'react';
import { H1, H2, LinkButton, Headshot, ExternalLinkButton } from 'components';
import headshot from 'images/headshot.jpeg';
import styled, { keyframes } from 'styled-components';

const Home = () => (
  <HomePageContainer id="homePage" data-testid="homePage">
    <Header>
      <H1 id="homePageHeader" data-testid="skillsPageHeader">
        Hello, I am Sean!
      </H1>
      <H2 id="homePageSubHeader" data-testid="homePageSubHeader">
        Software Engineer, Front End Web Developer, Jazz Musician{' '}
      </H2>
    </Header>
    <Main>
      <Headshot
        src={headshot}
        alt="headshotImage"
        id="headshotImage"
        data-testid="headshotImage"
      />
      <ButtonsContainer>
        <LinkButton route="/contact" title="Contact Me" />
        <ExternalLinkButton
          href="https://www.musicianthatcodes.com/"
          target="blank"
        >
          My Blog
        </ExternalLinkButton>
      </ButtonsContainer>
    </Main>
  </HomePageContainer>
);

const fadeInAnimation = keyframes`
    0% { 
        opacity: 0; 
    } 
    100% { 
        opacity: 1; 
    } 
`;

const HomePageContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
  transition: 0.5s;
  animation: ${fadeInAnimation} ease 1s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  @media (max-width: 975px) {
    flex-direction: column;
    justify-content: flex-start;
  }
`;

const Header = styled.header`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  @media (max-width: 975px) {
    align-items: center;
  }
`;

const Main = styled.main`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 1rem 0;
`;

const ButtonsContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

export default Home;
