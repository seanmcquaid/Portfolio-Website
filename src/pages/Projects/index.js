import React from 'react';
import { HeaderText } from 'components/components';
import styled from 'styled-components';
import projectsList from './projectsList';
import Project from './Project';

const Projects = () => (
  <ProjectsPageContainer id="projectsPage" data-testid="projectsPage">
    <Header id="projectsPageHeader" data-testid="projectsPageHeader">
      <HeaderText>Projects</HeaderText>
    </Header>
    <Main>
      <ProjectsList id="projectsList" data-testid="projectsList">
        {projectsList.map((project, i) => (
          <Project
            key={i}
            title={project.title}
            image={project.image}
            description={project.description}
            repoLink={project.repoLink}
            projectId={i}
          />
        ))}
      </ProjectsList>
    </Main>
  </ProjectsPageContainer>
);

const ProjectsPageContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
`;

const Header = styled.header`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 1rem;
`;

const Main = styled.main`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
`;

const ProjectsList = styled.ul`
  margin: 0;
  padding: 0;
  display: grid;
  list-style: none;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  overflow: auto;
  max-height: 32rem;
  @media (max-width: 975px) {
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    overflow: visible;
  }
`;

export default Projects;
