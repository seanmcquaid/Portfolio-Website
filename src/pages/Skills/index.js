import React from 'react';
import { HeaderText, ParagraphText } from 'components';
import styled from 'styled-components';
import skillsList from './skillsList';
import Skill from './Skill';

const Skills = () => (
  <SkillsPageContainer id="skillsPage" data-testid="skillsPage">
    <Header>
      <HeaderText id="skillsPageHeader" data-testid="skillsPageHeader">
        Skills
      </HeaderText>
      <ParagraphText>
        I currently work with Java/Springboot at work, however, my speciality is
        in Full-Stack JavaScript development. I specifically have a passion for
        creating visually appealing applications that solve a specific business
        problem. I currently spend most of my time outside of work attempting to
        come up with creative solutions in React and React Native.
      </ParagraphText>
    </Header>
    <Main>
      <SkillsList id="skillsList" data-testid="skillsList">
        {skillsList.map((skill, i) => {
          const { name, IconComponent } = skill;
          return <Skill key={i} IconComponent={IconComponent} name={name} />;
        })}
      </SkillsList>
    </Main>
  </SkillsPageContainer>
);

const SkillsPageContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
  @media (max-width: 975px) {
    justify-content: flex-start;
    flex-direction: column;
  }
`;

const Header = styled.header`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  max-width: 34rem;
  padding: 1rem;
  @media (max-width: 975px) {
    align-items: center;
  }
`;

const Main = styled.main`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const SkillsList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  max-width: 26rem;
  list-style: none;
  margin: 0;
  padding: 0;
`;

export default Skills;
