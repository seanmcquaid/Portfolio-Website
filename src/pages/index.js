import About from './About';
import Home from './Home';
import Contact from './Contact';
import Projects from './Projects';
import Skills from './Skills';
import Error from './Error';

export { About, Home, Contact, Projects, Skills, Error };
