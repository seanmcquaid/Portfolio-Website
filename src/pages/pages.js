import About from './About';
import Home from './Home';
import Contact from './Contact/Contact';
import Projects from './Projects/Projects';
import Skills from './Skills/Skills';
import Error from './Error';

export { About, Home, Contact, Projects, Skills, Error };
