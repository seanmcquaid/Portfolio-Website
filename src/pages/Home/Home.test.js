import React from "react";
import {render, cleanup, fireEvent, act, waitFor} from "@testing-library/react";
import {HashRouter as Router, Route} from "react-router-dom";
import {Home, Contact} from "pages/pages";
import "@testing-library/jest-dom/extend-expect";

const renderRoutes = () => (
    render(
        <Router>
            <Route exact path="/" component={Home}/>
            <Route exact path="/contact" component={Contact}/>
        </Router>
    )
);

describe("<Home/>", () => {

    afterEach(cleanup);
    it("Matches snapshot", () => {
        const home = render(<Router><Home/></Router>);
        expect(home).toMatchSnapshot();
    });

    it("Contact Me Button takes me to the Contact Page", async () => {
        jest.useFakeTimers();

        const {getByTestId} = renderRoutes();

        fireEvent.click(getByTestId("Contact MeLinkButton"));
    
        act(() => {
            jest.advanceTimersByTime(1600);
        });

        await waitFor(() => expect(() => getByTestId("loadingSpinner")).toThrowError());

        expect(getByTestId("contactPageHeader")).toBeVisible();
    });
});