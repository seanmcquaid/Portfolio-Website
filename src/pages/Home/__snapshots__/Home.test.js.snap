// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fznxKY dxythX"
        data-testid="homePage"
        id="homePage"
      >
        <div
          class="sc-fznMAR kdcijV"
        >
          <h1
            class="sc-fznxsB hwcVxy"
            data-testid="skillsPageHeader"
            id="homePageHeader"
          >
            Hello, I am Sean!
          </h1>
          <h2
            class="sc-fznJRM cQkQrk"
            data-testid="homePageSubHeader"
            id="homePageSubHeader"
          >
            Software Engineer, Front End Web Developer, Jazz Musician 
          </h2>
          <a
            class="sc-fznWqX kBttbi"
            data-testid="Contact MeLinkButton"
            href="#/contact"
          >
            Contact Me
          </a>
        </div>
        <div
          class="sc-fznWOq jtizem"
        >
          <img
            alt="headshotImage"
            class="sc-fzqARJ jmTNSy"
            data-testid="headshotImage"
            id="headshotImage"
            src="headshot.jpg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fznxKY dxythX"
      data-testid="homePage"
      id="homePage"
    >
      <div
        class="sc-fznMAR kdcijV"
      >
        <h1
          class="sc-fznxsB hwcVxy"
          data-testid="skillsPageHeader"
          id="homePageHeader"
        >
          Hello, I am Sean!
        </h1>
        <h2
          class="sc-fznJRM cQkQrk"
          data-testid="homePageSubHeader"
          id="homePageSubHeader"
        >
          Software Engineer, Front End Web Developer, Jazz Musician 
        </h2>
        <a
          class="sc-fznWqX kBttbi"
          data-testid="Contact MeLinkButton"
          href="#/contact"
        >
          Contact Me
        </a>
      </div>
      <div
        class="sc-fznWOq jtizem"
      >
        <img
          alt="headshotImage"
          class="sc-fzqARJ jmTNSy"
          data-testid="headshotImage"
          id="headshotImage"
          src="headshot.jpg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
